internal/checker/checker.go

```diff
@@ -17212,7 +17227,7 @@ func (c *Checker) getTypeFromBindingElement(element *ast.Node, includePatternInT
 		if ast.IsBindingPattern(element.Name()) {
 			contextualType = c.getTypeFromBindingPattern(element.Name(), true /*includePatternInType*/, false /*reportErrors*/)
 		}
-		return c.addOptionality(c.widenTypeInferredFromInitializer(element, c.checkDeclarationInitializer(element, CheckModeNormal, contextualType)))
```

Comment:
this new function is called from a couple of other places in Strada. Why not in Corsa too?

Reply:
I only see 3 `getWidenedLiteralTypeForInitializer` hits in both codebases 🤔 And they seem to be in the same places. Same for `widenTypeInferredFromInitializer` (4 hits)

Reply:
it splits the old `widenTypeInferredFromInitializer` into `widenTypeInferredFromInitializer` and `getWidenedLiteralTypeForInitializer`. You have the second call to `getWidenedLiteralTypeForInitializer` at the top of the new version of `widenTypeInferredFromInitializer`

Reply:
No, I'm wrong, it seems fine